account-compression/programs/account-compression/src/lib.rs

```diff
@@ -246,6 +257,7 @@ pub mod spl_account_compression {
 
         let header = ConcurrentMerkleTreeHeader::try_from_slice(header_bytes)?;
         header.assert_valid_authority(&ctx.accounts.authority.key())?;
+        header.assert_is_batch_initialized()?;
```

Comment:
Can you add a unit test that checks that `append_canopy_nodes` and `finalize_merkle_tree_with_root` cannot be called on batch initialized trees?

Reply:
I believe the `it('Should fail to double finalize the tree', async () => {` and `it('Should fail to replace a canopy node for a finalised tree', async () => {` are already covering the case for a finalized tree. For a regular (not batch-initialized) tree added some more tests